tests/Fixtures/Controller/BatchOtherController.php

```diff
@@ -18,6 +18,9 @@
 
 final class BatchOtherController
 {
+    /**
+     * @param ProxyQueryInterface<object> $aCustomNameForTheProxyQuery
```

Comment:
This is what I had in mind, I think it would be nice to be in docs too.

Reply:
That's the same idea that when you're doing `Collection<Foo>` or `AdminInterface<Foo>`. 

But there is no documentation in Doctrine, or Symfony or Sonata about generics. I don't think that the place for it.